### PR TITLE
Call into V8 from Rust

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

**Comment by robbym** [comment]
Use `impl Drop for Deno` instead of explicit `destroy`?

**Comment by ry** [comment]
@piscisaureus PTAL. This adds some more bindings to rust and calls into V8. Next up will be to call into the load system from the prototype - but serialization infra needs to come first. So I think we should just land this as is.

File: src/main.rs [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
+
+#[repr(C)]
+struct deno_buf {
+    data: *const c_char,
+    len: c_int, // TODO(ry) should be size_t.
+}
+
+#[repr(C)]
+struct DenoC {
+    _unused: [u8; 0],
```
Comment (piscisaureus) [inline src/main.rs]: What's this?

File: src/main.rs [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
 use libc::c_int;
+use libc::c_void;
 use std::ffi::CStr;
 use std::ffi::CString;
+use std::ptr;
+
+#[repr(C)]
+struct deno_buf {
+    data: *const c_char,
+    len: c_int, // TODO(ry) should be size_t.
```
Comment (piscisaureus) [inline src/main.rs]: In JS, (typed)arrays can't be longer than UINT32_MAX. So maybe this is appropriate.

File: src/main.rs (hunk 1 above)
Comment (ry) [reply src/main.rs]: This is a how you do an anonymous struct. In C this would be

‘struct DenoC;’

**Pull request merged**